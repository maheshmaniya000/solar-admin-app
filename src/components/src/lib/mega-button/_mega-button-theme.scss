@use '~@angular/material' as mat;

@mixin theme($theme-or-color-config) {
  $typography-config: mat.get-typography-config($theme-or-color-config);
  $color-config: mat.get-color-config($theme-or-color-config);
  $background: map-get($color-config, background);
  $primary: map-get($color-config, primary);

  sg-mega-button  {
    button.mat-button {
      background-color: mat.get-color-from-palette($background, hover);

      .mat-button-wrapper {
        color: mat.get-color-from-palette($primary);
        @include mat.typography-level($typography-config, subheading-2);
      }
    }
  }
}
