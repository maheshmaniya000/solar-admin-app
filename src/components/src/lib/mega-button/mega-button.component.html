<button mat-button>
  <img [src]="imageSrc"/>
  <ng-content></ng-content>
</button>
