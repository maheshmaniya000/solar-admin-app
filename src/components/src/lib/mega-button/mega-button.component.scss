@use '~@angular/material' as mat;

:host {
  button.mat-button {
    height: 72px;
    width: 100%;
    min-width: 160px;
    padding: 12px 8px 12px 12px;

    ::ng-deep {
      .mat-button-wrapper {
        display: flex;
        align-items: center;
        justify-content: left;
        text-align: left;
        margin-right: 12px;
        white-space: initial;
        overflow: hidden;
        height: 100%;

        img {
          width: 48px;
          height: 48px;
          margin-right: 12px;
          flex-shrink: 0;
        }
      }
    }
  }
}


