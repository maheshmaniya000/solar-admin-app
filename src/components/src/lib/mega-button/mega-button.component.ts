import { ChangeDetectionStrategy, Component, Input } from '@angular/core';

import { MegaButtonIcon } from './mega-button-icon.enum';

@Component({
  selector: 'sg-mega-button',
  templateUrl: './mega-button.component.html',
  styleUrls: ['./mega-button.component.scss'],
  changeDetection: ChangeDetectionStrategy.OnPush
})
export class MegaButtonComponent {
  imageSrc: string;

  @Input()
  set icon(icon: MegaButtonIcon) {
    this.imageSrc = `assets/img/mega-button/sprite.svg#${icon}`;
  }
}
