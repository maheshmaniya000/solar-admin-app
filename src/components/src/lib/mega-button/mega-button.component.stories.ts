import { Meta, moduleMetadata, Story } from '@storybook/angular';

import { SgDefaultsModule } from '../defaults/defaults.module';
import { MegaButtonIcon } from './mega-button-icon.enum';
import { MegaButtonComponent } from './mega-button.component';
import { MegaButtonModule } from './mega-button.module';

const styles = [
  `
  .container {
      display: grid;
      grid-template-columns: repeat(3, minmax(180px, 250px));
      column-gap: 1rem;
      row-gap: 1rem;
    }
  `
];

const allIcons = Object.keys(MegaButtonIcon).map(key => MegaButtonIcon[key]).sort();

export default {
  title: 'Solargis/Components/Mega Button',
  component: MegaButtonComponent,
  decorators: [
    moduleMetadata({
      imports: [SgDefaultsModule, MegaButtonModule]
    })
  ],
  argTypes: {
    onClick: { action: 'clicked', table: { disable: true }, control:false },
    allIcons: { table: { disable: true } }
  },
  args: {
    allIcons
  }
} as Meta;

export const standard: Story = props => ({
  props,
  template: `<sg-mega-button [icon]="icon" (click)="onClick()">{{label}}</sg-mega-button>`,
  styles
});

export const all: Story = props => ({
  props,
  template: `
      <div class="container">
        <sg-mega-button *ngFor="let icon of allIcons" [icon]="icon">{{icon}}</sg-mega-button>
      </div>`,
  styles
});

const argTypes = {
  icon: {
    name: 'Icon',
    options: allIcons,
    control: { type: 'select' },
    defaultValue: MegaButtonIcon['app-prospect']
  },
  label: {
    name: 'Label',
    control: { type: 'text' },
    defaultValue: 'Prospect Professional'
  }
};

standard.argTypes = argTypes;
