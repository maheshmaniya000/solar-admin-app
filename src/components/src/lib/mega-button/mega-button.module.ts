import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { MatButtonModule } from '@angular/material/button';

import { MegaButtonComponent } from './mega-button.component';

@NgModule({
  declarations: [MegaButtonComponent],
  imports: [CommonModule, MatButtonModule],
  exports: [MegaButtonComponent]
})
export class MegaButtonModule {}
