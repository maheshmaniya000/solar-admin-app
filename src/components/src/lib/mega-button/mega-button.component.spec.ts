import { createHostFactory, SpectatorHost } from '@ngneat/spectator/jest';

import { MegaButtonIcon } from './mega-button-icon.enum';
import { MegaButtonComponent } from './mega-button.component';

describe('MegaButtonComponent', () => {
  describe('unit', () => {
    let component: MegaButtonComponent;

    beforeEach(() => {
      component = new MegaButtonComponent();
    });

    describe('imageSrc', () => {
      it(`should return correct image src`, () => {
        component.icon = MegaButtonIcon['app-prospect'];
        expect(component.imageSrc).toEqual('assets/img/mega-button/sprite.svg#app-prospect');
      });
    });
  });

  describe('component', () => {
    let spectator: SpectatorHost<MegaButtonComponent>;
    const createHost = createHostFactory(MegaButtonComponent);

    beforeEach(
      () =>
        (spectator = createHost('<sg-mega-button [icon]="icon">Label</sg-mega-button>', {
          hostProps: {
            icon: MegaButtonIcon['app-prospect']
          }
        }))
    );

    it('should render mega button with correct icon', () => {
      expect(spectator.fixture).toMatchSnapshot();
    });
  });
});
