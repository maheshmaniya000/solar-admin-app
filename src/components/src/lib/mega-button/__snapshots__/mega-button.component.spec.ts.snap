// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MegaButtonComponent component should render mega button with correct icon 1`] = `
<ng-component
  __ngContext__={[Function LRootView]}
  icon={[Function String]}
>
  <sg-mega-button>
    <button
      mat-button=""
    >
      <img
        src="assets/img/mega-button/sprite.svg#app-prospect"
      />
      Label
    </button>
  </sg-mega-button>
</ng-component>
`;
